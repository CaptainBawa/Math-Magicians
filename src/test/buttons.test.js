import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import Buttons from '../Components/buttonsComponents';

describe('Buttons', () => {
  test('renders correctly', () => {
    const { container } = render(
      <Buttons type="button" value="Click me" className="primary" handleClick={() => {}} />,
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  test('calls handleClick function when clicked', () => {
    const handleClick = jest.fn();
    const { getByText } = render(
      <Buttons type="button" value="Click me" className="primary" handleClick={handleClick} />,
    );
    fireEvent.click(getByText('Click me'));
    expect(handleClick).toHaveBeenCalled();
  });
});
