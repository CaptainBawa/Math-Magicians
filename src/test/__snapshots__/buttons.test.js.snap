// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Buttons renders correctly 1`] = `
<button
  class="btn primary"
  type="button"
>
  Click me
</button>
`;
